### PR TITLE
fix(sigint): wait for all workers to stop upon Ctrl+C

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: src/cli.ts [hunk 1]
```diff
@@ -179,6 +184,8 @@ function addRunnerOptions(program: commander.Command, param: boolean) {
       .option('-g, --grep <grep>', 'Only run tests matching this string or regexp', '.*')
       .option('--global-timeout <timeout>', 'Specify maximum time this test suite can run (in milliseconds), default: 0 for unlimited', '0')
       .option('-j, --jobs <jobs>', 'Number of concurrent jobs, use 1 to run in single worker, default: (number of CPU cores / 2)', String(Math.ceil(require('os').cpus().length / 2)))
+      .option('--list', 'Only collect all the test and report them')
```
Comment (dgozman) [inline src/cli.ts:187]: There is also '--list' at line 200.

File: test/max-failures.spec.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      for (let i = 0; i < 10; ++i) {
+        it('fail_' + i, () => {
+          expect(true).toBe(false);
+        });
+      }
+    `
+  }, { 'max-failures': 8 });
+  expect(result.exitCode).toBe(1);
+  expect(result.failed).toBe(8);
+  expect(result.output.split('\n').filter(l => l.includes('expect(')).length).toBe(16);
```
Comment (dgozman) [inline test/max-failures.spec.ts:38]: I'm curious why 16 instead of 8?

File: src/workerRunner.ts [hunk 3]
```diff
@@ -138,31 +142,33 @@ export class WorkerRunner extends EventEmitter {
...
-    testInfo.duration = monotonicTime() - startTime;
-    if (this._testInfo) {
-      // We could have reported end due to an unhandled exception.
-      this.emit('testEnd', buildTestEndPayload(testId, testInfo));
-    }
-    if (!this._stopped && testInfo.status !== 'passed') {
+    if (timedOut && this._testInfo.status === 'passed')
+      this._testInfo.status = 'timedOut';
+    this._testInfo.duration = monotonicTime() - startTime;
+    // We could have reported end due to an unhandled exception.
```
Comment (dgozman) [inline src/workerRunner.ts:167]: IIUC, this comment is no longer relevant?

File: src/runner.ts [hunk 4]
```diff
@@ -116,17 +115,14 @@ export class Runner {
...
-      this._reporter.onBegin(config, suite);
-      await runner.run();
-      await runner.stop();
-      this._reporter.onEnd();
-    } finally {
-      for (const f of this._afterFunctions)
-        await f();
-    }
+    let sigint = false;
+    const sigIntPromise = new Promise(f => process.on('SIGINT', () => { sigint = true; f(); }));
```
Comment (dgozman) [inline src/runner.ts:119]: nit: might as well remove the sigint handler here, if we aim at reusable runner class.

**Review by dgozman — APPROVED** [review]

File: test/max-failures.spec.ts (hunk 2 above)
Comment (pavelfeldman) [reply test/max-failures.spec.ts:38]: 1) expect(true) failed
   ... source snippet expect(true)

**Pull request merged**